source/puppet/4.8/env_isolation.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: default
+title: "Isolating resources in multiple environments"
+canonical: "/puppet/latest/environment_isolation.html"
+description: "Generating metadata to isolate resources in environments in Puppet"
+---
+
+[code_mgr_env]: {{pe}}/code_mgr.html#environment-isolation-metadata
+
+
+If you have multiple environments, environment isolation prevents resource types from leaking between your various environments. To use environment isolation, you'll generate metadata files that Puppet can use in place of the normal Ruby resource type implementations.
```

Comment:
This comment might apply to a few other places in this doc as well (like the next section).